src/include/sof/drivers/esai.h

```diff
@@ -49,5 +49,137 @@
+#define ESAI_SAISR_TDE		BIT(15)
+#define ESAI_SAISR_TUE		BIT(14)
+#define ESAI_SAISR_TFS		BIT(13)
+#define ESAI_SAISR_RODF		BIT(10)
+#define ESAI_SAISR_REDF		BIT(9)
+#define ESAI_SAISR_RDF		BIT(8)
+#define ESAI_SAISR_ROE		BIT(7)
+#define ESAI_SAISR_RFS		BIT(6)
+#define ESAI_SAISR_IF2		BIT(2)
+#define ESAI_SAISR_IF1		BIT(1)
+#define ESAI_SAISR_IF0		BIT(0)
+
+#define ESAI_SAICR_ALC		BIT(8)
+#define ESAI_SAICR_TEBE		BIT(7)
+#define ESAI_SAICR_SYN		BIT(6)
+#define ESAI_SAICR_OF2		BIT(2)
+#define ESAI_SAICR_OF1		BIT(1)
+#define ESAI_SAICR_OF0		BIT(0)
+
+#define ESAI_xCR_xLIE		BIT(23)
+#define ESAI_xCR_xIE		BIT(22)
+#define ESAI_xCR_xEDIE		BIT(21)
+#define ESAI_xCR_xEIE		BIT(20)
+#define ESAI_xCR_xPR		BIT(19)
+#define ESAI_xCR_PADC		BIT(17) // tx only
+#define ESAI_xCR_xFSR		BIT(16)
+#define ESAI_xCR_xFSL		BIT(15)
+#define ESAI_xCR_xSWS_MASK	MASK(14, 10)
+#define ESAI_xCR_xSWS_VAL(s, w)	((w) < 24 ? ((s) - (w) + (((w) - 8) >> 2)) : ((s) < 32 ? 0x1e : 0x1f))
+#define ESAI_xCR_xSWS(s, w)	SET_BITS(14, 10, ESAI_xCR_xSWS_VAL(s, w))
+#define ESAI_xCR_xMOD_MASK	MASK(9, 8)
+#define ESAI_xCR_xMOD_NORMAL	SET_BITS(9, 8, 0)
+#define ESAI_xCR_xMOD_NETWORK	SET_BITS(9, 8, 1)
+#define ESAI_xCR_xMOD_AC97	SET_BITS(9, 8, 3)
+#define ESAI_xCR_xWA		BIT(7)
+#define ESAI_xCR_xSHFD		BIT(6)
+#define ESAI_xCR_TE_MASK	MASK(5, 0)
+#define ESAI_xCR_TE_SET(v)	SET_BITS(5, 0, v)
+#define ESAI_xCR_TE(v)		ESAI_xCR_TE_SET(MASK((v) - 1, 0))
+#define ESAI_xCR_RE_MASK	MASK(3, 0)
+#define ESAI_xCR_RE_SET(v)	SET_BITS(3, 0, v)
+#define ESAI_xCR_RE(v)		ESAI_xCR_RE_SET(MASK((v) - 1, 0))
+
+#define ESAI_xCCR_xHCKD		BIT(23)
+#define ESAI_xCCR_xFSD		BIT(22)
+#define ESAI_xCCR_xCKD		BIT(21)
+#define ESAI_xCCR_xHCKP		BIT(20)
+#define ESAI_xCCR_xFSP		BIT(19)
+#define ESAI_xCCR_xCKP		BIT(18)
+#define ESAI_xCCR_xFP_MASK	MASK(17, 14)
+#define ESAI_xCCR_xFP(div)	SET_BITS(17, 14, (div) - 1)
+#define ESAI_xCCR_xDC_MASK	MASK(13, 9)
+#define ESAI_xCCR_xDC(v)	SET_BITS(13, 9, (v) - 1)
+#define ESAI_xCCR_xDC_AC97	SET_BITS(13, 9, 0x0C)
+#define ESAI_xCCR_xPSR_MASK	MASK(8, 8)
+#define ESAI_xCCR_xPSR_BYPASS	SET_BITS(8, 8, 1)
+#define ESAI_xCCR_xPSR_DIV8	SET_BITS(8, 8, 0)
+#define ESAI_xCCR_xPM_MASK	MASK(7, 0)
+#define ESAI_xCCR_xPM(v)	SET_BITS(7, 0, (v) - 1)
+
+#define ESAI_xSMA_xS_MASK	MASK(15, 0)
+#define ESAI_xSMA_xS(v)		SET_BITS(15, 0, v)
+#define ESAI_xSMB_xS_MASK	MASK(15, 0)
+#define ESAI_xSMB_xS(v)		SET_BITS(15, 0, (v) >> 16)
+
+#define ESAI_PRRC_PDC_MASK	MASK(11, 0)
+#define ESAI_PRRC_PDC(v)	SET_BITS(11, 0, v)
+
+#define ESAI_PCRC_PC_MASK	MASK(11, 0)
+#define ESAI_PCRC_PC(v)		SET_BITS(11, 0, v)
+
+#define ESAI_GPIO		MASK(11, 0)
+
+#define EDMA_ESAI_IRQ		442
+
+#define EDMA_ESAI_TX_CHAN	7
+#define EDMA_ESAI_RX_CHAN	6
+
+#define trace_esai(format, ...) trace_event(TRACE_CLASS_DAI, format, ##__VA_ARGS__)
+#define tracev_esai(format, ...) tracev_event(TRACE_CLASS_DAI, format, ##__VA_ARGS__)
+#define trace_esai_error(format, ...) trace_error(TRACE_CLASS_DAI, format, ##__VA_ARGS__)
```

Comment:
```
#include <sof/trace/trace.h>
#include <user/trace.h>
```

Reply:
Same comment about ordering, is `<user/*>` more global than `<sof/*>`?

Reply:
Found another comment related to the ordering in a different PR, guess the answer is "yes".